airflow/providers/google/cloud/hooks/cloud_sql.py

```diff
@@ -407,6 +408,51 @@ def _wait_for_operation_to_complete(self, project_id: str, operation_name: str)
 )
 
 
+class CloudSQLAsyncHook(GoogleBaseAsyncHook):
+    """Class to get asynchronous hook for Google Cloud SQL."""
+
+    sync_hook_class = CloudSQLHook
+
+    async def get_conn(self, session, url):
+
+        scopes = [
+            "https://www.googleapis.com/auth/cloud-platform",
+            "https://www.googleapis.com/auth/sqlservice.admin",
+        ]
+
+        async with Token(scopes=scopes) as token:
+            session_aio = AioSession(session)
+
+            headers = {
+                "Authorization": f"Bearer {await token.get()}",
+            }
+
+            try:
+                operation = await session_aio.get(url=url, headers=headers)
+            except AirflowException:
+                pass
+
+        return operation
+
+    async def get_operation_name(self, project_id: str, operation_name: str, session):
+        url = f"https://sqladmin.googleapis.com/sql/v1beta4/projects/{project_id}/operations/{operation_name}"
+        return await self.get_conn(url=str(url), session=session)
+
+    async def get_operation(self, project_id: str, operation_name: str):
+        async with ClientSession() as session:
+            try:
+                operation = await self.get_operation_name(
+                    project_id=project_id,
+                    operation_name=operation_name,
+                    session=session,
+                )
+                operation = await operation.json(content_type=None)
+            except Exception as e:
```

Comment:
would it be possible to catch the error related to this API only? The exception looks very broad. Also since will are trying to read some param from this exception trigger so reducing scope make sense to me